### PR TITLE
fix: allow using interface defs from imported modules

### LINKED ISSUE #2900: Importing interfaces not as docs describe
### Version Information

* vyper Version (output of `vyper --version`): 0.3.3
* OS: osx
* Python Version (output of `python --version`): 3.9.5

### What's your issue about?

Please include information like:

* full output of the error you received
* what command you ran
* the code that caused the failure (see [this link](https://help.github.com/articles/basic-writing-and-formatting-syntax/) for help with formatting code)
* please try running your example with the --debug flag turned on

I have an interface that looks like such:

```python
# @version ^0.3.3
interface IFace:
    def read_stuff() -> uint256: view
```

And a contract that looks as such:

```python
# @version ^0.3.3
from interfaces.IFace import IFace

@external
@view
def read_contract(some_address: address) -> uint256:
    myContract: IFace = IFace(some_address)
    return myContract.read_stuff()
```

However, when I run `vyper contracts/MyContract.vy` I get:

```
Error compiling: contracts/MyContract.vy
vyper.exceptions.UnknownAttribute: IFace has no member 'read_stuff'. 
  contract "contracts/MyContract.vy", function "read_contract", line 11:11 
       10     myContract: IFace = IFace(some_address)
  ---> 11     return myContract.read_stuff()
  -------------------^
       12
```

[According to the documentation](https://vyper.readthedocs.io/en/stable/interfaces.html?highlight=interface#importing-interfaces), when importing interfaces, you need to use regular vyper syntax. However, adjusting my interface as such:

```python
# @version ^0.3.3

interface IFace:
    @external
    def read_stuff() -> uint256: 
        pass
```

Results in this error:

```
vyper.exceptions.StructureException: Body must only contain state mutability label
  contract "contracts/interfaces/IFace.vy", function "read_stuff", line 6:8 
       5     def read_stuff() -> uint256:
  ---> 6         pass
  ---------------^
       7
```

### How can it be fixed?

I'm not sure

### PR DESCRIPTION
**Pull request by charles-cooper**
### What I did
fix #2900 

### How to verify it
see new test

### Commit message

```
- add interface defs to ModuleT's exposed `get_type_members()`
- slight refactor of ModuleT to have a special helper instead of
dispatching to `self.interface`
```

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: tests/functional/codegen/modules/test_interface_imports.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    contract = """
+import ifaces
+
+@external
+def test_foo(s: ifaces.IFoo) -> bool:
+    assert s.foo() == block.number
+    return True
+    """
+
+    input_bundle = make_input_bundle({"ifaces.vy": ifaces})
```
Comment (pcaversaccio) [inline tests/functional/codegen/modules/test_interface_imports.py:26]:
```suggestion
    input_bundle = make_input_bundle({"ifaces.vyi": ifaces})
```

**Review by fubuloubu — APPROVED** [review]

**Review by pcaversaccio — COMMENTED** [review]
The unit test could be enriched with another test using a relative import from the parent directory (i.e. `from ..parent.ifaces.IFoo import IFoo`) just to ensure that the `interface` keyword doesn't break path resolution in some weird way.

File: tests/functional/codegen/modules/test_interface_imports.py (hunk 1 above)
Comment (charles-cooper) [reply tests/functional/codegen/modules/test_interface_imports.py:26]: going to skip this since i want to phase out `.vyi` files anyway

**Comment by pcaversaccio** [comment]
> > The unit test could be enriched with another test using a relative import from the parent directory (i.e. `from ..dir.ifaces.IFoo import IFoo`) just to ensure that the `interface` keyword doesn't break path resolution in some weird way.
> 
> in vyper you can't use `from` to import types inside of files. like in python, you can do `from my_module import some_type`. but in vyper you can't import individual names from a module -- the usage here would be `from ..dir import ifaces; ... ; ifaces.IFoo`.

FWIW, I still use that syntax in snekmate: https://github.com/pcaversaccio/snekmate/blob/main/src%2Ftokens%2FERC20.vy#L75

**Comment by charles-cooper** [comment]
> FWIW, I still use that syntax in snekmate: https://github.com/pcaversaccio/snekmate/blob/main/src%2Ftokens%2FERC20.vy#L75

yea as discussed offline -- that syntax was probably a bug, and it does not work anymore on master (probably as of #3663 ) anyways.

**Comment by pcaversaccio** [comment]
> > FWIW, I still use that syntax in snekmate: https://github.com/pcaversaccio/snekmate/blob/main/src%2Ftokens%2FERC20.vy#L75
> 
> yea as discussed offline -- that syntax was probably a bug, and it does not work anymore on master (probably as of #3663 ) anyways.

Yup, amended in snekmate via https://github.com/pcaversaccio/snekmate/pull/194.

**Pull request merged**